### PR TITLE
u-boot: introduce git and release versions for u-boot-mainline

### PR DESCRIPTION
**Pull request by jorisoffouga**
Signed-off-by: Joris Offouga <offougajoris@gmail.com>

### CODE REVIEW

File: recipes-bsp/u-boot/u-boot-stm-common.inc [hunk 1]
```diff
@@ -13,9 +13,9 @@ SRC_URI = " \
 	file://0002-stm32mp1-Enable-redundant-environment.patch \
 "
 
-SRCREV = "${AUTOREV}"
```
Comment (texierp) [inline recipes-bsp/u-boot/u-boot-stm-common.inc:16]: NOK, the file `u-boot-stm-common.inc` is a common header.
Please move SRCREV to `u-boot-mainline_2019.10.bb`, ex:

```
require recipes-bsp/u-boot/u-boot.inc
require u-boot-stm-common.inc

SRCREV = "61ba1244b548463dbfb3c5285b6b22e7c772c5bd"
```

Also
, you have to create a new file `u-boot-mainline_git.bb` with the following contents
for instance

```
require recipes-bsp/u-boot/u-boot.inc
require u-boot-stm-common.inc

SRCREV = "${AUTOREV}"
DEFAULT_PREFERENCE = "-1"
```

So, the common header is defined as follows:

```
# Copyright (C) 2019 Koncepto
# Released under the MIT license (see COPYING.MIT for the terms)

DESCRIPTION = "U-Boot Mainline to test stm32mp1"
LICENSE = "GPLv2+"
LIC_FILES_CHKSUM = "file://Licenses/README;md5=30503fd321432fc713238f582193b78e"

DEPENDS += "bison-native dtc-native"
PROVIDES += "u-boot"

SRC_URI = " \
	git://gitlab.denx.de/u-boot/u-boot.git;protocol=https \
	file://0001-stm32mp1-Disable-ADC.patch \
	file://0001-stm32mp1-Replace-ENV_IS_EXT4-to-MMC.patch \
	file://0002-stm32mp1-Enable-redundant-environment.patch \
"

S = "${WORKDIR}/git"

UBOOT_MAKE_TARGET_append = " u-boot-initial-env"

do_install_append () {
	install -d ${D}/${sysconfdir}
	install -m 644 ${B}/u-boot-initial-env ${D}/${sysconfdir}/u-boot-initial-env
}

do_deploy_append () {
	install -d ${DEPLOYDIR}
	install -m 644 ${B}/u-boot-initial-env ${DEPLOYDIR}/u-boot-initial-env
}

PACKAGES =+ "${PN}-initial-env"
FILES_${PN}-initial-env = "${sysconfdir}/u-boot-initial-env"
PACKAGE_ARCH = "${MACHINE_ARCH}"
COMPATIBLE_MACHINE = "(stm*)"
```

Not tested ;)

**Review by texierp — CHANGES_REQUESTED** [review]

File: recipes-bsp/u-boot/u-boot-stm-common.inc (hunk 1 above)
Comment (jorisoffouga) [reply recipes-bsp/u-boot/u-boot-stm-common.inc:16]: Ok i will do it ;)

**Comment by texierp** [comment]
Also, my suggestion for the commit title is:

`u-boot: introduce git and release versions for u-boot-mainline`

**Comment by texierp** [comment]
Reviewed-by: Pierre-Jean Texier <pjtexier@koncepto.io>

**Pull request merged**